### PR TITLE
Use AppInsights instead of System.Diagnostics.Tracing in DiagnosticsService

### PR DESCRIPTION
**Pull request by xavierdecoster**
Summary of the changes:

* `DiagnosticsService` is not tracing through `System.Diagnostics.TraceSource` anymore but directly through Application Insights `TelemetryClient.TrackTrace` API
* Kept `Trace.AutoFlush = true` for now, as I believe we still want that. This setting does cause the AI Trace Listener to `Flush()` each trace through AI separately, which previously lead to lock contention on flushing the buffer too often. However, the intended change of this PR is to no longer trace through `System.Diagnostics.Trace`, so the rare events that are traced through `System.Diagnostics.Trace` may as well be flushed immediately. However, we should be less exposed to any potential perf impact introduced by auto-flushing behavior of AI Trace Listener.

This change should loosen our dependency on `Microsoft.ApplicationInsights.TraceListener`, although I'm not removing it completely. 
* Some trace events, e.g. during `WebRole.OnStop` still go through `System.Diagnostics.Trace`, and (external) dependencies may be using the `System.Diagnostics.Trace` APIs. 
* We also still wire up the [`DiagnosticMonitorTraceListener`](https://docs.microsoft.com/en-us/azure/cloud-services/cloud-services-dotnet-diagnostics-trace-flow) in `AppActivator`, [line 113](https://github.com/NuGet/NuGetGallery/blob/f2c21be69b34e3f8fc1cc192d4e1ca09bcc1a9d5/src/NuGetGallery/App_Start/AppActivator.cs#L113). 

If you think I missed a spot, or have concerns, let's discuss!

Addresses #7512

### CODE REVIEW

File: src/NuGetGallery/App_Start/AppActivator.cs [hunk 1]
```diff
@@ -39,6 +39,8 @@ public static class AppActivator
 
         public static void PreStart()
         {
+            Trace.AutoFlush = true;
```
Comment (joelverhagen) [inline src/NuGetGallery/App_Start/AppActivator.cs]: Do we really need this? This seems a bit scary.

File: src/NuGetGallery.Core/Diagnostics/IDiagnosticsSource.cs [hunk 2]
```diff
@@ -12,10 +11,10 @@ namespace NuGetGallery.Diagnostics
     public interface IDiagnosticsSource : ILogger
     {
         void ExceptionEvent(Exception exception);
-        
-        void TraceEvent(TraceEventType type, int id, string message,
+
+        void TraceEvent(LogLevel logLevel, EventId eventId, string message,
```
Comment (joelverhagen) [inline src/NuGetGallery.Core/Diagnostics/IDiagnosticsSource.cs]: This is kind of the wrong interface for `Microsoft.Extensions.Logging`. `ILogger` works better when you don't combine the parameters with the message yourself. We can punt this part if you want but it shouldn't be too bad if you add an `params object[]` parameter.

File: src/NuGetGallery/App_Start/AppActivator.cs (hunk 1 above)
Comment (xavierdecoster) [reply src/NuGetGallery/App_Start/AppActivator.cs]: We already had it [here](https://github.com/NuGet/NuGetGallery/pull/7513/files#diff-0b70d5e18960c2ab4f8b24f71be12222L18), I just moved it around to application pre-start outside of the constructor of `DiagnosticsService`.

Happy to remove it completely, but it's hard to determine if we miss telemetry by removing this.

File: src/NuGetGallery.Core/Diagnostics/IDiagnosticsSource.cs (hunk 2 above)
Comment (xavierdecoster) [reply src/NuGetGallery.Core/Diagnostics/IDiagnosticsSource.cs]: Perhaps that work should be done as part of a separate work item. Want to log an issue for that and describe the change you'd like to see? This PR just changes the type of the first two arguments on this particular interface method.

File: src/NuGetGallery/App_Start/AppActivator.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGetGallery/App_Start/AppActivator.cs]: Oh, okay. My mistake.

File: src/NuGetGallery.Services/Diagnostics/TraceDiagnosticsSource.cs [hunk 3]
```diff
@@ -111,20 +87,7 @@ bool ILogger.IsEnabled(LogLevel logLevel)
 
         IDisposable ILogger.BeginScope<TState>(TState state)
```
Comment (skofman1) [inline src/NuGetGallery.Services/Diagnostics/TraceDiagnosticsSource.cs]: I assume this is not used?

File: src/NuGetGallery.Core/Diagnostics/IDiagnosticsSource.cs (hunk 2 above)
Comment (joelverhagen) [reply src/NuGetGallery.Core/Diagnostics/IDiagnosticsSource.cs]: I'll leave it be for now. If we find this painful in the future we can open an issue. Thanks for clarifying.

File: src/NuGetGallery.Services/Diagnostics/TraceDiagnosticsSource.cs (hunk 3 above)
Comment (xavierdecoster) [reply src/NuGetGallery.Services/Diagnostics/TraceDiagnosticsSource.cs]: I think not. Scopes are not supported in AI traces AFAIK, so `NotSupportedException` seems appropriate?

Comment (agr) [reply src/NuGetGallery.Services/Diagnostics/TraceDiagnosticsSource.cs]: Logging shouldn't throw, should it?

Comment (xavierdecoster) [reply src/NuGetGallery.Services/Diagnostics/TraceDiagnosticsSource.cs]: @agr What's your suggested way to implement this unsupported API?

**Pull request merged**